Merge agenda improvements

ahamdi opened the pull request.

Telgou committed:
Improvment : Adding download button and logic for ics file - EXO-73522 (#673)

* Corrected the ics generation function, added translation for the event's description
* Added conditional display based on a FeatureFlag

Telgou committed:
fix: ics generation if there is no webconference url - EXO-73523 (#674)

* Improvment : Adding download button for ICS file - EXO-73522

* Corrected the ics generation function, added translation for the event's description

* Added conditional display based on FeatureFlag

* fix: ics generation if there is no conference url - EXO-73523

Telgou committed:
fix: ics generation for google calendar - EXO-73684 (#675)

Added specific field required by Google Calendar to import an ICS file

Telgou committed:
fix: ics generation with plain text description - EXO-73523 (#677)

Added the description field for plain text, for Calendar clients that do not use HTML description

Telgou committed:
Improvment: Connect personal calendar option - EXO-73528 (#682)



    Ensured that agenda connectors options are only displayed when there is at least one connector enabled by the admin.
    Updated the UI of the right column of the event details to match the current state of agenda connectors settings from both the user and administration sides.

Telgou committed:
Improvment: Make the title of the event mandatory and has at least 1 char - EXO-73525 (#685)

Telgou committed:
Improvment: Rich text editor for the events description- EXO-73526 (#687)

 - Added rich editor to the event description.
 - Ensured the new formatted description displays correctly when viewed on the platform and in programs after importing the ICS file.
 - Added try-catch block in email notif template to handle cases where the conference variable is not binded to ensure that the template does not throw a MissingPropertyException.

Telgou committed:
Improvment: UI review of the events details body - EXO-73527 (#688)

 - Conducted UI review for the agenda event details body
 - Better ICS file generation on the server side and eventtitle only as subject if creator or updater

ahamdi committed:
fix: add AltRep field for event description - EXO-73523 (#690)

ahamdi committed:
fix: adjust buttons for ICS and external agenda connectors - EXO-73528 (#694)

ahamdi committed:
fix: set the size of right column - EXO-73527 (#695)

ahamdi committed:
fix: remove redundent event title from email notifications - EXO-73531 (#696)

ahamdi committed:
fix: Update event form UI - EXO-73527  (#697)

ahamdi committed:
fix: disable tag suggestion from Agenda event description - EXO-73526  (#698)

* fix: disable tag suggestion from Agenda event description - EXO-73526

* fix: fix the style of the Mail notification - EXO-73526

ahamdi committed:
fix: Update event form UI - EXO-73527 (#703)

ahamdi committed:
fix: Hide Download ICS CTA button when an Agenda connector is connected - EXO-73528 (#705)

When a remote agenda connector is connected, the CTA button Add top my Agenda should be hidden

ahamdi committed:
fix: Add tooltip for new buttons - EXO-73528 (#706)

ahamdi committed:
fix: Description shouldn't be centered on Date polls - EXO-73527 (#704)

ahamdi committed:
fix: adapt ics to be imported by main agenda clients - EXO-73523 (#707)

In this fix :

    removed the plain text description
    removed the unneeded properties DESCRIPTION;ALTREP
    Added support to Emoticons in description
    Fixed the encoding of accentuated characters

ahamdi committed:
fix: start and end dates are rset to midnight - EXO-73523 (#708)

ahamdi committed:
fix: display correctly italic text in poll description - EXO-73526 (#709)

ahamdi committed:
fix: add location to ics attached to emails - EXO-73977 (#710)

This fix adds the Location attribute to the ics file generated by the server when sending email notifications.

ahamdi committed:
fix: remove avatar from the email notifications of Agenda - EXO-74507 (#711)

ahamdi committed:
fix: load web conferening data when sycnhronizing events with Connectors - EXO-62884 (#712)

The fix loads web conferencing information when loading events to make sure that information is sent when the synchronization with personal is done using the Agenda connectors

ahamdi committed:
fix: generate ics content from a single place - EXO-73977

rdenarie commented on agenda-services/src/main/java/org/exoplatform/agenda/rest/AgendaEventRest.java:
```diff
@@ -1713,4 +1714,78 @@ public Response getPendingEvents(
+          )
+          String timeZoneId) {
+    try {
+      if (eventId <= 0) {
```
How do you check the current user have access to this event ? 

As a user I must'nt be able to get the ICS of an event I'm not in.

ahamdi replied:
The function  agendaEventService.getEventById(eventId, timeZoneId, identityId) checks if the user has access to the event otherwise throws an IllegalAccessException https://github.com/exoplatform/agenda/blob/develop/agenda-api/src/main/java/org/exoplatform/agenda/service/AgendaEventService.java#L64

rdenarie replied:
yes but, in 
`Identity targetUserIdentity = identityManager.getIdentity(String.valueOf(identityId));`

identityId is provided as parameter. So I can provide the id I want, and I can provide an ID which is not mine.

ahamdi replied:
@rdenarie Yes to get an event information, you will need to provide a valid event id with the identity ID of a participant. I will remove completely the identity ID in this case, and I will get the current user from the request context, WDYT ?

rdenarie commented on agenda-services/src/main/java/org/exoplatform/agenda/util/NotificationUtils.java:
```diff
@@ -286,15 +283,6 @@ public static final void setNotificationRecipients(IdentityManager identityManag
     String showParticipants = getFullUserName(participants, identityManager);
     notification.with(STORED_PARAMETER_EVENT_ATTENDEES, showParticipants);
 
-    // After computing all usernames, to whom, notifications will be sent,
-    // this deletes the username of modifier/creator user
-    long userIdentityToExclude = StringUtils.equals(typeModification, "ADDED") ? event.getCreatorId() : modifierId;
-    if (userIdentityToExclude > 0) {
-      Identity identityToExclude = identityManager.getIdentity(String.valueOf(userIdentityToExclude));
-      if (identityToExclude != null) {
-        recipients.remove(identityToExclude.getRemoteId());
-      }
-    }
```
Why do you remove this part ?
It prevent user modifier to receive a notification about an event he just modify

ahamdi replied:
This change was requested in task https://community.exoplatform.com/portal/dw/tasks/taskDetail/73531

rdenarie replied:
ok.

ahamdi committed:
fix:allow users to generate events just for their events - EXO-73977

Rest endpint will not take the identity ID as parameter and it will use the one associated with the current user to avoid that
users can generate ics of events that they know their attendees identity Ids .

exo-swf committed:
Merge Translations

rdenarie commented on agenda-services/src/main/java/org/exoplatform/agenda/rest/AgendaEventRest.java:
```diff
@@ -1713,4 +1715,75 @@ public Response getPendingEvents(
+      if (StringUtils.isBlank(timeZoneId)) {
+        timeZoneId = TimeZone.getDefault().getID();
+      }
+      Identity currentUserIdentity = this.identityManager.getOrCreateUserIdentity(request.getRemoteUser());
```
Should it be better to use the new way to get current user : 
https://community.exoplatform.com/portal/dw/activity?id=445564
?

ahamdi replied:
@rdenarie Here I am getting just the username of the current user from the HTTP request and wants to check if the user exists, and did not use the user ACL since I do not check permissions. Could you give more details about your suggestion ?

rdenarie replied:
My bad, I misunderstood the content of [community.exoplatform.com/portal/dw/activity?id=445564](https://community.exoplatform.com/portal/dw/activity?id=445564)

What is better between ConversationState.getCurrent and request.getRemoteUser ?

ahamdi replied:
@rdenarie IMO request.getRemoteUser is better as it is based on the HTTP servlets  standard. ConversationState.getCurrent is based on ThreadLocal which is a specific implementation of eXo core, and may need to change when we will go stateless .
